### PR TITLE
ci: add tofu ci checks

### PR DESCRIPTION
**Pull request by MxNxPx**
add some basic tofu CI

### CODE REVIEW

File: modules/subnet/main.tf [hunk 1]
```diff
@@ -12,7 +12,7 @@ resource "nutanix_subnet" "subnet" {
   vlan_id     = var.vlan_id
   subnet_type = "VLAN"
 
-  prefix_length = 24
+  prefix_length = var.prefix_length
```
Comment (MxNxPx) [inline modules/subnet/main.tf:15]: `prefix_length` was hardcoded rather than using the var

File: modules/subnet/variables.tf [hunk 2]
```diff
@@ -25,7 +25,7 @@ variable "gateway_ip" {
 variable "prefix_length" {
   description = "The prefix length of the subnet, ex: 24"
   type        = number
-  default     = null
+  default     = 24
```
Comment (MxNxPx) [inline modules/subnet/variables.tf:28]: `prefix_length` wasn't being used at all, so when updated to pull this in, set a default

File: modules/rke2/outputs.tf [hunk 3]
```diff
@@ -3,5 +3,5 @@
 
 output "bootstrap_ip" {
   description = "IP address for the bootstrap virtual machines."
-  value       = length(nutanix_virtual_machine.rke2_bootstrap) > 0 ? nutanix_virtual_machine.rke2_bootstrap[0].nic_list_status.0.ip_endpoint_list[0]["ip"] : null
+  value       = length(nutanix_virtual_machine.rke2_bootstrap) > 0 ? nutanix_virtual_machine.rke2_bootstrap[0].nic_list_status[0].ip_endpoint_list[0]["ip"] : null
```
Comment (MxNxPx) [inline modules/rke2/outputs.tf:6]: this was found by `tflint`, with this output:

`Warning - List items should be accessed using square brackets (terraform_deprecated_index)`

File: modules/rke2/main.tf [hunk 4]
```diff
@@ -166,7 +166,7 @@ resource "nutanix_virtual_machine" "rke2_agents" {
     authorized_keys  = var.ssh_authorized_keys,
     node_user        = var.node_user,
     token            = var.join_token,
-    connect_hostname = var.server_dns_name != "" ? var.server_dns_name : nutanix_virtual_machine.rke2_bootstrap[0].nic_list_status.0.ip_endpoint_list[0]["ip"],
+    connect_hostname = var.server_dns_name != "" ? var.server_dns_name : nutanix_virtual_machine.rke2_bootstrap[0].nic_list_status[0].ip_endpoint_list[0]["ip"],
```
Comment (MxNxPx) [inline modules/rke2/main.tf:169]: this was found by `tflint`, with this output:

`Warning - List items should be accessed using square brackets (terraform_deprecated_index)`

File: modules/postgres-profile-vm/outputs.tf [hunk 5]
```diff
@@ -3,5 +3,5 @@
 
 output "bootstrap_ip" {
   description = "IP address for the bootstrap virtual machines."
-  value       = nutanix_virtual_machine.postgres_profile_vm.nic_list_status.0.ip_endpoint_list[0]["ip"]
+  value       = nutanix_virtual_machine.postgres_profile_vm.nic_list_status[0].ip_endpoint_list[0]["ip"]
```
Comment (MxNxPx) [inline modules/postgres-profile-vm/outputs.tf:6]: this was found by `tflint`, with this output:

`Warning - List items should be accessed using square brackets (terraform_deprecated_index)`

**Review by jacobbmay — APPROVED** [review]
LGTM

**Pull request merged**